src/main/java/com/pingcap/tikv/types/DataTypeFactory.java

```diff
@@ -69,7 +69,7 @@
           .put(TYPE_NEW_DECIMAL, DecimalType::new)
           .put(TYPE_FLOAT, RealType::new)
           .put(TYPE_DOUBLE, RealType::new)
-          .put(TYPE_DURATION, TimestampType::new)
+          .put(TYPE_DURATION, DurationType::new)
```

Comment:
okay, i will push another pr for this

Reply:
DuratingType is misleading.  Part of Duration was taken card in TimestampType. Either we rename this class or we can remove Duration in TimestampType and add it into DurationType.  What do you think?

Reply:
`DurationType` is named after its previous implementation in TiDB and its flag type `TYPE_DURATION`. In this pr it has no relation with `TimestampType` anymore by implementing `IntegerType` instead.

Reply:
I mean this. https://github.com/pingcap/tikv-client-lib-java/blob/master/src/main/java/com/pingcap/tikv/types/TimestampType.java#L86

Reply:
This `decode()` function here is removed because of unreachable logic, we will call `decode()` unless we HAVE found a `NULL_FLAG`, which contradicts with logic here.

Reply:
Oh. I did not mean not. I mean `TimestampType` decode `Duration` at the lone of code. In this PR, you create a `DurationType`. My point is it is better move "decode Duration" login into this newly created class `DurationType`. Again, I am so story for the misunderstanding.

Reply:
I get your point, so you are suggesting Duration decoding in `TimestampType` ambiguous. I agree with your point. As a matter of fact, I don't see the `INT_FLAG` usage involving `TimestampType` either. There aren't any tests considering duration or any logic that leads to this condition clause. I will delete the usage and do integration test again.

Reply:
Yes. You are right. This part logic was not used. It added simply because we try to decode all time related type in this one class. In reality, it does not work out. So it is better remove it and add it into somewhere else.